pymess/backend/dialer/daktela.py

```diff
@@ -0,0 +1,100 @@
+from chamber.utils.datastructures import ChoicesEnum
+from django.utils import timezone
+from django.utils.encoding import force_text
+from django.utils.translation import ugettext_lazy as _l
+
+from pymess.backend.dialer import DialerBackend
+from pymess.config import settings
+from pymess.models import DialerMessage
+from pymess.utils.logged_requests import generate_session
+
+
+class DaktelaDialerBackend(DialerBackend):
+    """
+    Dialer backend implementing Daktela service https://www.daktela.com/api/v6/models/campaignsrecords
+    """
+
+    STATE = ChoicesEnum(
+        ('NOT_ASSIGNED', _l('not assigned'), 'not assigned'),
+        ('READY', _l('ready'), 'ready'),
+        ('RESCHEDULED_BY_DIALER', _l('rescheduled by dialer'), 'rescheduled by dialer'),
+        ('CALL_IN_PROGRESS', _l('call in progress'), 'call in progress'),
+        ('HANGUP', _l('hangup'), 'hangup'),
+        ('DONE', _l('done'), 'done'),
+        ('RESCHEDULED', _l('rescheduled'), 'rescheduled'),
+    )
+
+    # TODO: clean-up mappings
+    STATES_MAPPING = {
+        STATE.NOT_ASSIGNED: DialerMessage.STATE.UNKNOWN,
+        STATE.READY: DialerMessage.STATE.READY,
+        STATE.RESCHEDULED_BY_DIALER: DialerMessage.STATE.RESCHEDULED,
+        STATE.CALL_IN_PROGRESS: DialerMessage.STATE.IN_PROGRESS,
+        STATE.HANGUP: DialerMessage.STATE.DONE,
+        STATE.DONE: DialerMessage.STATE.DONE,
+        STATE.RESCHEDULED: DialerMessage.STATE.RESCHEDULED,
+    }
+
+    @staticmethod
+    def _get_dialer_api_url(name=None):
+        base_url = '/'.join((
+            settings.DIALER_API_URL,
+            settings.DIALER_API_VERSION,
+            settings.DIALER_API_RESOURCE,
+        ))
+        name = '/' + name if name else ''
+
+        return '{base_url}{name}.json?accessToken={access_token}'.format(
+            base_url=base_url, name=name, access_token=settings.DIALER_API_ACCESS_TOKEN,
+        )
+
+    def _update_dialer_states(self, messages):
+        """
+        Method uses Daktela API to get info about autodialer call status
+        :param messages: list of dialer messages to update
+        """
+        for message in messages:
+            client_url = self._get_dialer_api_url(message.tag)
+            result = generate_session(slug='pymess - Daktela autodialer', related_objects=(message,)).get(client_url)
```

Comment:
ten slug dělej podle pravidel slugů v djangu "instance that ensures a value consists of only letters, numbers, underscores or hyphens."

Reply:
spravil som to podla toho, ako som to videl inde v tejto kniznici ;)